prow/plugins/dco/dco.go

```diff
@@ -202,7 +203,18 @@ func checkExistingLabels(gc gitHubClient, l *logrus.Entry, org, repo string, num
 // takeAction will take appropriate action on the pull request according to its
 // current state.
 func takeAction(gc gitHubClient, cp commentPruner, l *logrus.Entry, org, repo string, pr github.PullRequest, commitsMissingDCO []github.RepositoryCommit, existingStatus string, hasYesLabel, hasNoLabel, addComment bool) error {
-	targetURL := fmt.Sprintf("https://github.com/%s/%s/blob/master/CONTRIBUTING.md", org, repo)
+	// Get default branch for repo to construct link to CONTRIBUTING.md (default: master)
+	repoObj, err := gc.GetRepo(org, repo)
```

Comment:
This will have a terrible cache hit rate, as the response contains details like `pushed_at`, number of stars and similiar. Can we build a cached version of this in `prow/github`?

Reply:
"this" being "GetDefaultBranch(org, repo)"

Reply:
I was wondering whether grabbing the entire repo object every time would be detrimental. Yes, I'll work on improving that today.

Reply:
Sorry for the delay. Implementing a cache is a bit new for me, but I've implemented a basic in-memory cache for `GetDefaultBranch` in `client.go`. Please take a look when you have a moment! Thanks!